phpstan-baseline.neon

```diff
@@ -0,0 +1,1027 @@
+			count: 1
+			path: lib/Doctrine/DBAL/Schema/Visitor/SchemaDiffVisitor.php
+
+		-
+			message: "#^Class Doctrine\\\\DBAL\\\\Statement implements generic interface IteratorAggregate but does not specify its types\\: TKey, TValue$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Statement.php
+
+		-
+			message: "#^Property Doctrine\\\\DBAL\\\\Statement\\:\\:\\$stmt type has no value type specified in iterable type Doctrine\\\\DBAL\\\\Driver\\\\Statement\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Statement.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Statement\\:\\:getIterator\\(\\) return type has no value type specified in iterable type Traversable\\<mixed, mixed\\>\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Statement.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Statement\\:\\:getWrappedStatement\\(\\) return type has no value type specified in iterable type Doctrine\\\\DBAL\\\\Driver\\\\Statement\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Statement.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Tools\\\\Console\\\\Command\\\\ImportCommand\\:\\:configure\\(\\) has no return typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Tools/Console/Command/ImportCommand.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Tools\\\\Console\\\\Command\\\\ReservedWordsCommand\\:\\:configure\\(\\) has no return typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Tools/Console/Command/ReservedWordsCommand.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Tools\\\\Console\\\\Command\\\\RunSqlCommand\\:\\:configure\\(\\) has no return typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Tools/Console/Command/RunSqlCommand.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Tools\\\\Console\\\\ConsoleRunner\\:\\:printCliConfigTemplate\\(\\) has no return typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Tools/Console/ConsoleRunner.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Types\\\\ConversionException\\:\\:conversionFailedSerialization\\(\\) has no return typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Types/ConversionException.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Types\\\\ConversionException\\:\\:conversionFailedSerialization\\(\\) has parameter \\$error with no typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Types/ConversionException.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Types\\\\ConversionException\\:\\:conversionFailedSerialization\\(\\) has parameter \\$format with no typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Types/ConversionException.php
+
+		-
+			message: "#^Method Doctrine\\\\DBAL\\\\Types\\\\ConversionException\\:\\:conversionFailedSerialization\\(\\) has parameter \\$value with no typehint specified\\.$#"
+			count: 1
+			path: lib/Doctrine/DBAL/Types/ConversionException.php
```

Comment:
Having this many untriaged errors will complicate the project maintenance in the near future:

1. If someone makes a contribution and sees a new error, how do they know if they should fix it or whitelist? If triaged, some of the non-fixable issues could be generalized and cover the future code changes.
2. PHPStan on `2.10` doesn't cover `tests/` while it does on `master`. Once `2.10` is attempted to be merged to `master`, we'll can see potentially similar number of issues in `tests/`. How are we going to handle that?

Reply:
> If someone makes a contribution and sees a new error, how do they know if they should fix it or whitelist? If triaged, some of the non-fixable issues could be generalized and cover the future code changes.

New errors on new code must be fixed.

> PHPStan on 2.10 doesn't cover tests/ while it does on master. Once 2.10 is attempted to be merged to master, we'll can see potentially similar number of issues in tests/. How are we going to handle that?

I'd baseline existing (yet undiscovered) errors.

Reply:
> New errors on new code must be fixed.

That's the problem. They may have the same underlying root cause as the other errors that are baselined.

> I'd baseline existing (yet undiscovered) errors.

That's why I'm asking to minimize the number of such errors.

Reply:
Yes, @morozov is totally right. Baseline only works if new code can be written the correct way. So the level should be lowered or special care should be given to manually ignore common errors (that we want to ignore).

Reply:
I'm solving the issues, so there's no need for us to keep discussing this :smile:

Reply:
> Yes, @morozov is totally right.

What's wrong about this kind of discussion?

Reply:
I was just excited that someone gets the baseline feature as it was designed :D